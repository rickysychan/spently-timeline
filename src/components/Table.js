import React from 'react'
import PropTypes from 'prop-types'
import { convertTime } from '../helper'

const Table = ({ email, emailType, timestamp }) => 
    <table className="table"> 
        <tbody>
            <tr className="table__row">
                <th>Email</th>
                <th>Email Type</th>
                <th>Time Stamp</th>
            </tr>
            <tr className="table-data">
                <td className="table-email">{email}</td>
                <td className="table-emailType">{emailType}</td>
                <td className="table-timestamp">{convertTime(timestamp)}</td>
            </tr>
        </tbody>
    </table>

Table.propTypes = {
    email: PropTypes.string.isRequired,
    emailType: PropTypes.string.isRequired,
    timestamp: PropTypes.number.isRequired
}

Table.defaultProps = {
    email: 'moo@gmail.com',
    emailType: 'mooo',
    timestamp: 123
}


export default Table