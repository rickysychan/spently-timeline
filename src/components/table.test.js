import React from 'react'
import { shallow } from 'enzyme'
import Table from './Table'

const table = shallow(<Table />)

describe('Table', () => {
    it('should render properly', () => {
        expect(table).toMatchSnapshot()
    })
})