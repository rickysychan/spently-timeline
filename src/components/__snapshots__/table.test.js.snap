// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    email="moo@gmail.com"
    emailType="mooo"
    timestamp={123}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <tbody>
        <tr
                className="table__row"
        >
                <th>
                        Email
                </th>
                <th>
                        Email Type
                </th>
                <th>
                        Time Stamp
                </th>
        </tr>
        <tr
                className="table-data"
        >
                <td
                        className="table-email"
                >
                        moo@gmail.com
                </td>
                <td
                        className="table-emailType"
                >
                        mooo
                </td>
                <td
                        className="table-timestamp"
                >
                        Dec-31-1969 19:02:03
                </td>
        </tr>
</tbody>,
      "className": "table",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <tr
            className="table__row"
>
            <th>
                        Email
            </th>
            <th>
                        Email Type
            </th>
            <th>
                        Time Stamp
            </th>
</tr>,
          <tr
            className="table-data"
>
            <td
                        className="table-email"
            >
                        moo@gmail.com
            </td>
            <td
                        className="table-emailType"
            >
                        mooo
            </td>
            <td
                        className="table-timestamp"
            >
                        Dec-31-1969 19:02:03
            </td>
</tr>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                Email
</th>,
              <th>
                Email Type
</th>,
              <th>
                Time Stamp
</th>,
            ],
            "className": "table__row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email Type",
              },
              "ref": null,
              "rendered": "Email Type",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Time Stamp",
              },
              "ref": null,
              "rendered": "Time Stamp",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <td
                className="table-email"
>
                moo@gmail.com
</td>,
              <td
                className="table-emailType"
>
                mooo
</td>,
              <td
                className="table-timestamp"
>
                Dec-31-1969 19:02:03
</td>,
            ],
            "className": "table-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "moo@gmail.com",
                "className": "table-email",
              },
              "ref": null,
              "rendered": "moo@gmail.com",
              "type": "td",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mooo",
                "className": "table-emailType",
              },
              "ref": null,
              "rendered": "mooo",
              "type": "td",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dec-31-1969 19:02:03",
                "className": "table-timestamp",
              },
              "ref": null,
              "rendered": "Dec-31-1969 19:02:03",
              "type": "td",
            },
          ],
          "type": "tr",
        },
      ],
      "type": "tbody",
    },
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tbody>
          <tr
                    className="table__row"
          >
                    <th>
                              Email
                    </th>
                    <th>
                              Email Type
                    </th>
                    <th>
                              Time Stamp
                    </th>
          </tr>
          <tr
                    className="table-data"
          >
                    <td
                              className="table-email"
                    >
                              moo@gmail.com
                    </td>
                    <td
                              className="table-emailType"
                    >
                              mooo
                    </td>
                    <td
                              className="table-timestamp"
                    >
                              Dec-31-1969 19:02:03
                    </td>
          </tr>
</tbody>,
        "className": "table",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr
              className="table__row"
>
              <th>
                            Email
              </th>
              <th>
                            Email Type
              </th>
              <th>
                            Time Stamp
              </th>
</tr>,
            <tr
              className="table-data"
>
              <td
                            className="table-email"
              >
                            moo@gmail.com
              </td>
              <td
                            className="table-emailType"
              >
                            mooo
              </td>
              <td
                            className="table-timestamp"
              >
                            Dec-31-1969 19:02:03
              </td>
</tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Email
</th>,
                <th>
                  Email Type
</th>,
                <th>
                  Time Stamp
</th>,
              ],
              "className": "table__row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email Type",
                },
                "ref": null,
                "rendered": "Email Type",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Time Stamp",
                },
                "ref": null,
                "rendered": "Time Stamp",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  className="table-email"
>
                  moo@gmail.com
</td>,
                <td
                  className="table-emailType"
>
                  mooo
</td>,
                <td
                  className="table-timestamp"
>
                  Dec-31-1969 19:02:03
</td>,
              ],
              "className": "table-data",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "moo@gmail.com",
                  "className": "table-email",
                },
                "ref": null,
                "rendered": "moo@gmail.com",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "mooo",
                  "className": "table-emailType",
                },
                "ref": null,
                "rendered": "mooo",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dec-31-1969 19:02:03",
                  "className": "table-timestamp",
                },
                "ref": null,
                "rendered": "Dec-31-1969 19:02:03",
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
